### PR TITLE
FAT-1004 - Created Karate test for Loans

### PR DESCRIPTION
**Pull request by sherzodn**
## Purpose
[FAT-1004](https://issues.folio.org/browse/FAT-1004)
_When a loaned item is checked in at a service point that does not serve its location and no request exists, change the item status to In-transit and destination to primary service point for its location (circ-14)_

## Approach
- Reproduced test scenario with UI.
- Learned necessary sent requests and then necessary received responses 
- Based on the request/response created karate test scenario

## Learning
- learn karate framework from original doc file on the github
- looked other scenarios how they are described.

### CODE REVIEW

File: mod-circulation/src/main/resources/domain/mod-circulation/features/loans.feature [hunk 1]
```diff
@@ -232,7 +232,7 @@ Feature: Loans tests
     * call read('classpath:domain/mod-circulation/features/util/initData.feature@PostHoldings')
     * call read('classpath:domain/mod-circulation/features/util/initData.feature@PostItem') { extItemId: #(itemId), extItemBarcode: #(extItemBarcode)}
 
-    # post a user
+   # post a user
```
Comment (alexanderkurash) [inline mod-circulation/src/main/resources/domain/mod-circulation/features/loans.feature:235]: Indentation

Comment (sherzodn) [reply mod-circulation/src/main/resources/domain/mod-circulation/features/loans.feature]: done

File: mod-circulation/src/main/resources/domain/mod-circulation/features/loans.feature [hunk 2]
```diff
@@ -293,3 +293,54 @@ Feature: Loans tests
...
+    * def extLocationId1 = call uuid1
+    * def extLocationId2 = call uuid1
+
+    # first location and service point setup
+    * call read('classpath:domain/mod-circulation/features/util/initData.feature@PostLocation'){ extLocationId: #(extLocationId1),  extInstitutionId: #(extInstitutionId1), extCampusId: #(extCampusId1), extLibraryId: #(extLibraryId1), extServicePointId: #(extServicePointId1) }
+    * call read('classpath:domain/mod-circulation/features/util/initData.feature@PostServicePoint'){ servicePointId: #(extServicePointId1) }
+
+    # second location and service point setup
+    * call read('classpath:domain/mod-circulation/features/util/initData.feature@PostLocation'){ extLocationId: #(extLocationId2), extInstitutionId: #(extInstitutionId2), extCampusId: #(extCampusId2), extLibraryId: #(extLibraryId2), extServicePointId: #(extServicePointId2) }
+    * call read('classpath:domain/mod-circulation/features/util/initData.feature@PostServicePoint'){ servicePointId: #(extServicePointId2) }
```
[the remark this answers is not in the document]
Comment (sherzodn) [reply mod-circulation/src/main/resources/domain/mod-circulation/features/loans.feature]: done

File: mod-circulation/src/main/resources/domain/mod-circulation/features/loans.feature [hunk 3]
```diff
@@ -293,3 +293,54 @@ Feature: Loans tests
...
+    * def extLibraryId2 = call uuid1
+    * def extLocationId1 = call uuid1
+    * def extLocationId2 = call uuid1
+
+    # first location and service point setup
+    * call read('classpath:domain/mod-circulation/features/util/initData.feature@PostLocation'){ extLocationId: #(extLocationId1),  extInstitutionId: #(extInstitutionId1), extCampusId: #(extCampusId1), extLibraryId: #(extLibraryId1), extServicePointId: #(extServicePointId1) }
+    * call read('classpath:domain/mod-circulation/features/util/initData.feature@PostServicePoint'){ servicePointId: #(extServicePointId1) }
+
+    # second location and service point setup
+    * call read('classpath:domain/mod-circulation/features/util/initData.feature@PostLocation'){ extLocationId: #(extLocationId2), extInstitutionId: #(extInstitutionId2), extCampusId: #(extCampusId2), extLibraryId: #(extLibraryId2), extServicePointId: #(extServicePointId2) }
```
[the remark this answers is not in the document]
Comment (sherzodn) [reply mod-circulation/src/main/resources/domain/mod-circulation/features/loans.feature]: done

File: mod-circulation/src/main/resources/domain/mod-circulation/features/loans.feature [hunk 4]
```diff
@@ -293,3 +293,54 @@ Feature: Loans tests
...
+    * def extCampusId1 = call uuid1
+    * def extCampusId2 = call uuid1
+    * def extLibraryId1 = call uuid1
+    * def extLibraryId2 = call uuid1
+    * def extLocationId1 = call uuid1
+    * def extLocationId2 = call uuid1
+
+    # first location and service point setup
+    * call read('classpath:domain/mod-circulation/features/util/initData.feature@PostLocation'){ extLocationId: #(extLocationId1),  extInstitutionId: #(extInstitutionId1), extCampusId: #(extCampusId1), extLibraryId: #(extLibraryId1), extServicePointId: #(extServicePointId1) }
+    * call read('classpath:domain/mod-circulation/features/util/initData.feature@PostServicePoint'){ servicePointId: #(extServicePointId1) }
```
[the remark this answers is not in the document]
Comment (sherzodn) [reply mod-circulation/src/main/resources/domain/mod-circulation/features/loans.feature]: done

File: mod-circulation/src/main/resources/domain/mod-circulation/features/loans.feature [hunk 5]
```diff
@@ -293,3 +293,54 @@ Feature: Loans tests
...
+    * def extInstitutionId2 = call uuid1
+    * def extCampusId1 = call uuid1
+    * def extCampusId2 = call uuid1
+    * def extLibraryId1 = call uuid1
+    * def extLibraryId2 = call uuid1
+    * def extLocationId1 = call uuid1
+    * def extLocationId2 = call uuid1
+
+    # first location and service point setup
+    * call read('classpath:domain/mod-circulation/features/util/initData.feature@PostLocation'){ extLocationId: #(extLocationId1),  extInstitutionId: #(extInstitutionId1), extCampusId: #(extCampusId1), extLibraryId: #(extLibraryId1), extServicePointId: #(extServicePointId1) }
```
[the remark this answers is not in the document]
Comment (sherzodn) [reply mod-circulation/src/main/resources/domain/mod-circulation/features/loans.feature]: done

**Pull request merged**